python/dgl/graph_index.py

```diff
@@ -812,6 +812,56 @@ def from_scipy_sparse_matrix(self, adj):
         self._init(src, dst, edge_ids, num_nodes)
 
 
+    def from_csr_matrix(self, indptr, indices, edge_dir, shared_mem_name=""):
+        """Load a graph from the CSR matrix.
+
+        Parameters
+        ----------
+        indptr : a 1D tensor
+            index pointer in the CSR format
+        indices : a 1D tensor
+            column index array in the CSR format
+        edge_dir : string
+            the edge direction
+        shared_mem_name : string
+            the name of shared memory
+        """
+        assert self.is_readonly()
+        indptr = utils.toindex(indptr)
+        indices = utils.toindex(indices)
+        edge_ids = utils.toindex(F.arange(0, len(indices)))
+        self._handle = _CAPI_DGLGraphCSRCreate(
+            indptr.todgltensor(),
+            indices.todgltensor(),
+            edge_ids.todgltensor(),
+            shared_mem_name,
+            self._multigraph,
+            edge_dir)
```

Comment:
So `from_csr_matrix` actually *creates* a shared memory object if needed.

I feel it's better to have a utility function `create_shared_mem_from_csr_matrix()`, so we can replace this whole method with `from_shared_mem_csr_matrix`.

Reply:
not necessary. if `shared_mem_name` is given, csr is loaded to shared memory. otherwise, it's loaded to regular memory.